hubblestack/extmods/grains/hostuuid.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+'''
+Generate a unique uuid for this host, storing it on disk so it persists across
+restarts
+'''
+import logging
+import os
+import uuid
+
+log = logging.getLogger(__name__)
+
+
+def host_uuid():
+    '''
+    Generate a unique uuid for this host, storing it on disk so it persists
+    across restarts
+    '''
+    cached_uuid_path = os.path.join(os.path.dirname(__opts__['configfile']), 'hubble_cached_uuid')
+    existing_uuid = __opts__.get('host_uuid', None)
+    try:
+        if os.path.isfile(cached_uuid_path):
+            with open(cached_uuid_path, 'r') as f:
+                cached_uuid = f.read()
+                # Check if it's changed out from under us -- problem!
+                if existing_uuid and cached_uuid != existing_uuid:
+                    log.error('host_uuid changed on disk unexpectedly!'
+                              '\nPrevious: {0}\nNew: {1}\nKeeping previous.'
+                              .format(existing_uuid, cached_uuid))
+                    return {'host_uuid': existing_uuid}
```

Comment:
Trying to think about all possible outcomes, so set me straight: here we exit with returning the existing uuid. But when do we overwrite the cache file? If we don't overwrite, this fill keep logging every grains refresh.

Reply:
Good catch. I fixed it.